internal/status/updater_test.go

```diff
@@ -393,4 +395,59 @@ var _ = Describe("Updater", func() {
 			})
 		})
 	})
+
+	Describe("Skip GatewayClass updates", Ordered, func() {
+		var (
+			updater status.Updater
+			gc      *v1beta1.GatewayClass
+		)
+
+		BeforeAll(func() {
+			updater = status.NewUpdater(status.UpdaterConfig{
+				GatewayCtlrName:          gatewayCtrlName,
+				GatewayClassName:         gcName,
+				Client:                   client,
+				Logger:                   zap.New(),
+				Clock:                    fakeClock,
+				PodIP:                    "1.2.3.4",
+				UpdateGatewayClassStatus: false,
+			})
+
+			gc = &v1beta1.GatewayClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: gcName,
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "GatewayClass",
+					APIVersion: "gateway.networking.k8s.io/v1beta1",
+				},
+			}
+		})
+
+		It("should create resources in the API server", func() {
+			Expect(client.Create(context.Background(), gc)).Should(Succeed())
+		})
+
+		It("should not update GatewayClass status", func() {
+			updater.Update(
+				context.Background(),
+				state.Statuses{
+					GatewayClassStatus: &state.GatewayClassStatus{
+						ObservedGeneration: 1,
+						Conditions:         status.CreateTestConditions("Test"),
+					},
+				},
+			)
+
+			latestGc := &v1beta1.GatewayClass{}
+			expectedGc := gc.DeepCopy()
+
+			err := client.Get(context.Background(), types.NamespacedName{Name: gcName}, latestGc)
+			Expect(err).Should(Not(HaveOccurred()))
+
+			expectedGc.ResourceVersion = latestGc.ResourceVersion
+
+			Expect(helpers.Diff(expectedGc, latestGc)).To(BeEmpty())
```

Comment:
We could probably simplify this test and just check that latestGc status is not set, rather than comparing to another resource.